### PR TITLE
Auto Docstrings

### PR DESCRIPTION
**Pull request by rafaqz**
This works fine, but the existing bug with doctrings breaks the tests!

I made an attempt at fixing it but it might be better if someone who knows the codebase better tracks that bug down.

see: https://github.com/mauro3/Parameters.jl/issues/59

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -36,12 +36,12 @@ end
 @test MT1().c=="sdaf"
 @test "Test documentation\n" == Markdown.plain(@doc MT1)
 # https://github.com/JuliaLang/julia/issues/27092 means this does not work:
-# @test "A field\n" == Markdown.plain(@doc MT1.c)
+# @test "A field Default: sdaf\n" == Markdown.plain(@doc MT1.c)
 if VERSION<v"0.7-"
-    @test "A field\n" == Markdown.plain(Base.Docs.fielddoc(MT1, :c))
+    @test "A field Default: sdaf\n" == Markdown.plain(Base.Docs.fielddoc(MT1, :c))
```
Comment (mauro3) [inline test/runtests.jl]: Can you add a test for the docs of `r` too?

**Comment by mauro3** [comment]
Other than that one test, LGTM.

**Comment by rafaqz** [comment]
Don't pull yet I think I found some more bugs actually using it. Maybe I'll write a few more tests

**Comment by rafaqz** [comment]
Sorry forgot to get back to this! There were no more bugs it was something else of mine not handling docstrings. Good to go!

**Comment by rafaqz** [comment]
A tag for this would be good too

**Comment by mauro3** [comment]
Thanks!  If you are not in a hurry I will wait with the tag until the named-tuple stuff is done.  If that gets delayed and I forget, ping me.

**Pull request merged**